sharding/nodesCoordinator/sovereignIndexHashedNodesCoordinator.go

```diff
@@ -206,15 +215,97 @@ func (ihnc *sovereignIndexHashedNodesCoordinator) GetConsensusValidatorsPublicKe
 }
 
 // EpochStartPrepare is not implemented for sovereign
-func (ihnc *sovereignIndexHashedNodesCoordinator) EpochStartPrepare(_ data.HeaderHandler, _ data.BodyHandler) {
-	log.Error("sovereignIndexHashedNodesCoordinator.EpochStartPrepare was called, not implemented in sovereign")
+func (ihnc *sovereignIndexHashedNodesCoordinator) EpochStartPrepare(hdr data.HeaderHandler, body data.BodyHandler) {
+	if !hdr.IsStartOfEpochBlock() {
+		log.Error("could not process EpochStartPrepare on nodesCoordinator - not epoch start block")
```

Comment:
Maybe add `sovereign` in the log error messages here and below.

Reply:
+1

Reply:
Added it